### PR TITLE
LPS-154973 Update permission on bulk for several documents

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
Hi @liferay-usm we want you to review this.

With this PR we want to change the behavior of the permission tool, allowing us to change the permission for several documents at the same time. Currently only for documents.

This changes will allow various resourcePrimKey, so as to modify them through the permission modal.
As we have several, we will make a Map with the permission for each of the actions (by actionId), filling it in with the permission for group/company/etc, on top of the individual permission for each of the resources (on `PortletConfigurationPermissionsDisplayContext.populateResourcePermissionActionIds`).
Doing this we can check if each resource has each of the permissions or not. 
It is called indeterminate when a permission is missing a resource but present in another resource (`PortletConfigurationPermissionsDisplayContext.isIndeterminate`).
The delivery of the permission (via the checkbox) activated or deactivated does not change, however the indeterminate state does.
The checkbox states are sent for each permission. This is essential to check if we need to modify each resource permissions.
For this reason we have had to check this new indeterminate state. For each of the actions with the indeterminate state, we check the previous state of the resource to keep it. ( `​​PortletConfigurationPortlet._getIndeterminateStateActionIds` and `​​PortletConfigurationPortlet._getRoleIdsToActionIdsResourcePrimKey`)

For the indeterminate state we used a tri-state checkbox. At the beginning we applied the changes on CheckboxTaglib. 
A PR (https://github.com/liferay-frontend/liferay-portal/pull/2388) was sent to the Frontend Infra Team. They concluded that the behavior was very specific and it was better to create a separate component, so the `PermissionsCheckbox` was created.
In the mentioned PR the behavior of the component is explained in detail.
In a nutshell: the logic of the management toolbar checkbox was applied to this component. When it’s in the indeterminate state a specific value is sent, in this case `indeterminate` instead of the checkbox default value.

A known issue:
We don’t provide any workaround to show tooltips for over disabled elements because we are waiting for it on LPS-104333. If it isn’t implemented, we will provide an alternative solution before removing the feature flag.

To activate the feature the FF is `feature.flag.LPS-87806`

### CODE REVIEW

**Comment by boton** [comment]
ci:test:stable

**Comment by AliciaGarciaGarcia** [comment]
ci:test:stable

**Comment by drewbrokke** [comment]
@AliciaGarciaGarcia Thank you for sending this change to us! I will review it today and tomorrow and leave feedback.

**Comment by drewbrokke** [comment]
@AliciaGarciaGarcia thank you for your patience. I've made a number of refactor changes in a new PR here: https://github.com/liferay-usm/liferay-portal/pull/660. There should be no functional changes.

Please review and test this new branch to ensure that it still fulfills the requirements, and don't hesitate to leave feedback or concerns on the pull. 

Thanks!
